torch_geometric/graphgym/config.py

```diff
@@ -229,9 +229,15 @@ def set_cfg(cfg):
     # Evaluate model on test data every eval period epochs
     cfg.train.eval_period = 10
 
+    # Option to skip training epoch evaluation
```

Comment:
I think adding a configuration that selects the customized statistics could be helpful. Of course, we can also assume that all the registered customized statistics will be used. Please let me know if what do you think.

Reply:
Yeah, I think it's a good idea to specify which custom metric group to use.

In the [current setup](https://github.com/RemyLau/pytorch_geometric/blob/bed926723b5ef7a4e261522e5ba56e019c7e9917/torch_geometric/graphgym/logger.py#L163-L168), I tried to follow the GraphGym code design language for loading custom functions (the example I had in mind was the [``torch_geometric.graphgym.loadder.py``](https://github.com/RemyLau/pytorch_geometric/blob/bed926723b5ef7a4e261522e5ba56e019c7e9917/torch_geometric/graphgym/loader.py#L164-L168)):
1. Look through all relevant registered custom functions and return immediately if a custom function returns nontrivial results
2. If the above failed, use the default graphgym functions

Going back to the point of specifying which custom metric group to use, maybe one possible strategy could be:
1. Specify the desired custom metric group to use as ``cfg.custom_metric`` (set to ``'none'`` by default)
2. Instead of looking through all registered custom metric functions, find the one with the matched key of ``cfg.custom_metric``. And maybe print warning message if no matched key for ``cfg.custom_metric`` was found and ``cfg.custom_metric != 'none'`` and fall back to using the default implemented metrics.

Let me know what you think

Reply:
I think this API sounds great! You could further make `cfg.custom_metric` to be a list, since a user may want to use multiple customized metrics. If the list is empty, then no customized metric is used.